### PR TITLE
IN-38: Add support for logging and sentry.

### PR DESCRIPTION
**Pull request by dhirendrx**

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
-if (config.error) {
-  console.error(config.error);
+
+if (dotConfig.error) {
+  console.error(dotConfig.error);
 }
 
-module.exports = config.parsed;
\ No newline at end of file
+module.exports = config = Object.assign({}, dotConfig.parsed);
```
Comment (shalotelli) [inline lib/config.js]: why are you cloning the config?

File: lib/logging/sentry-transport.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    this.Sentry.configureScope(scope => {
+      Object.keys(extra).forEach((key) => {
+        scope.setExtra(key, extra[key]);
+      });
+      Object.keys(metadata).forEach((key) => {
+        scope.setExtra(key, metadata[key]);
+      });
+      this.Sentry.captureException(error);
+    });
+    return callback();
```
Comment (shalotelli) [inline lib/logging/sentry-transport.js:49]: above you've called this with `new SentryTransport(config.sentryDSN, {level: 'error'})` which means `callback` is `undefined`. would that throw an error here? maybe we should be more defensive here?

**Review by shalotelli — APPROVED** [review]
very minor request for change, didn't want to hold up the process so have approved.

File: lib/logging/sentry-transport.js (hunk 2 above)
Comment (dhirendra-linq) [reply lib/logging/sentry-transport.js:49]: The callback is passed to `log` function by `winston`.  It won't be `undefined`.

File: lib/config.js (hunk 1 above)
Comment (dhirendra-linq) [reply lib/config.js]: No reason.  will update it.

**Pull request merged**